### DISCUSSION TITLE
lua: print output to dap console? and howto read user input while debugging?

### DISCUSSION THREAD

**Discussion by qrockz** [mfussenegger/nvim-dap#1099] [Q&A]
### Howto print output to dap console?
 (**_nvim-dap-ui_**)
Currently every print statement appears into **_REPL window_**  but I want these print outputs directed into the **DAP console window**. How can I manage that?

### Howto read user input while debugging?
**_(local-lua-debugger-vscode)_**
I've red somewhere that input operations like` io.read(`) not working with lua debugger because the debugger itself uses this _io stream_ for communication. Am I wrong? (I hope so...plz)
Currently the debugger stops working whenever I trie to `io.read()` something  😟.

**Comment by mihaifm** [comment]
Basically having the same issues, but for a C program. Any luck solving them?

**Reply by qrockz** [reply]
@mihaifm unfortunately not

**Reply by frere-jacques** [reply]
Maybe this helps a bit:

For anybody searching how you can print to the dap-repl of nvim-dap instead of nvim repl.
```lua
local repl = require "dap.repl"
repl.append("Hello!")
```
I would prefer to print to the dapui_console instead, but, I couldn't find anything.

**Comment by MartyMcFlyInTheSky** [comment]
I'm having a similar issue where I can't see the output of `io.stdout:write("Hello")` from a debuggee inside the integrated terminal.

**Comment by aburto22** [comment]
Same here for a Javascript application. Most console logs go straight to REPL, and not console. For some reason "Jest" output (test runner) go to console though 🤔
